src/pyflexplot/s3.py

```diff
@@ -154,3 +144,44 @@ def _retry_with_backoff(fn: Callable,
             logging.info("Sleep: %.2f seconds", sleep)
             time.sleep(sleep)
             x += 1
+
+def _configure_s3_client(bucket):
+    """
+    Configures the S3 client based on the bucket's platform.
+
+    Args:
+        bucket: An object containing bucket configuration attributes.
+                Must include 'name', 'retries' and optionally 'endpoint_url', 's3_access_key', and 's3_secret_key'.
+
+    Returns:
+        A configured boto3 S3 client.
+    """
+    common_config = Config(
+        retries={
+            'max_attempts': int(bucket.retries),
+            'mode': 'standard'
+        }
+    )
+
+    # Check if endpoint_url is present to differentiate between AWS and other platforms
+    if hasattr(bucket, 'endpoint_url') and bucket.endpoint_url.strip():
+        # Non-AWS configuration
+        return boto3.Session().client(
+            's3',
+            endpoint_url=bucket.endpoint_url,
+            aws_access_key_id=bucket.s3_access_key,
+            aws_secret_access_key=bucket.s3_secret_key,
+            config=common_config
+        )
+    else:
+        # AWS S3 configuration
+        return boto3.Session().client(
+            's3',
+            config=Config(
+                region_name=bucket.region,
+                retries={
+                    'max_attempts': int(bucket.retries),
+                    'mode': 'standard'
+                }
+            )
```

Comment:
In this configuration for AWS S3, the region has been specified, whereas in the EWC configuration, the region name is absent. I now added `common_retries` instead of `common_config` to avoid repition between the two clients in this function.